src/new_subscr.cpp

```diff
@@ -3139,26 +3142,64 @@ void SW_MMap::draw(BITMAP* dest, int32_t xofs, int32_t yofs, SubscrPage& page) c
 					drawgrid(dest,tx+8,ty+8,c.dngn_fg,-1);
 				}
 			}
-			
+
 			break;
 		}
 		}
 	}
-	
+
+	if ((showVisRooms || showUnvisRooms) && type != dmOVERW) //broken for overworlds
+	{
+		auto si = 0;
+
+		for (int y2 = 0; y2 <= 7; ++y2)
+		{
+			byte dl = thedmap.grid[si];
+
+			for (int x2 = 0; x2 <= 7; ++x2)
+			{
+				if (!(dl & 0x80))
+				{
+					dl <<= 1;
+					continue;
+				}
+
+				dl <<= 1;
+
+				int scrx = x2 - thedmap.xoff;
```

Comment:
per [this commit](https://github.com/ZQuestClassic/ZQuestClassic/commit/75c45c19ac01d6dfed0049e4369b17388622ef1c), this is a bug. Should be:
```suggestion
				int scrx = x2 + thedmap.xoff;
```